### DISCUSSION TITLE
.env Variable not updated in CP

### DISCUSSION THREAD

**Discussion by dolphiq-allard** [craftcms/cms#10821] [Q&A] [answered]
Hey guys,

Ran into a small issue on my local development environment. The Craft CP is not reading the correct .env variable. It uses a value different than in the .env file.

I try to use the following variable(s):

`PRIMARY_SITE_URL_EN=http://sg.local:8080/`
`PRIMARY_SITE_URL=http://sg.local:8080/se/`

However as soon as I address the `PRIMARY_SITE_URL` variable in the CP it states `http://sg.local:8080/`. In fact I can delete the line from the .env and it still works. If I add a new variable this will become available in the CP instantly with the correct value. Furthermore, if I change the `PRIMARY_SITE_URL_EN` variable, this change **is** reflected in the CP.

Funny thing is it used to work just fine.
This is the first time I run into this problem. Is there some cache I need to clear maybe?

All help is highly appreciated.

Thanks much!

**Comment by brandonkelly** [comment, marked as answer]
Did you remember to include the `$` in `$PRIMARY_SITE_URL` when using that?

**Reply by dolphiq-allard** [reply]
Hey Brandon,

Thanks for the quick reply.

Yes, in the CP I call the environmental vars using `$`. See below for an example:

![image](https://user-images.githubusercontent.com/4312073/160137890-1a9de01d-e2a0-4360-a936-ae4b922b2907.png)

**Reply by brandonkelly** [reply]
Try removing `PRIMARY_SITE_URL` from `.env` entirely, then go to **Utilities** → **PHP Info** in the control panel, and search for `PRIMARY_SITE_URL` there. Do you see any occurrences of it (besides `PRIMARY_SITE_URL_EN`)?

**Reply by dolphiq-allard** [reply]
Removed it from the .env and searched for it, and it is present in de PHP Info stating the incorrect value.

**Reply by brandonkelly** [reply]
In that case, it must be getting defined deeper down, in your web server configuration. Which is why the `.env` changes aren’t having the intended effects.

**Reply by dolphiq-allard** [reply]
My bad, this was it. Had it defined in my Dockerfile as well. Thanks for pointing me in the right direction!
